crates/keystore/src/storage/substrate.rs

```diff
@@ -35,9 +57,21 @@ mod acco_sr25519 {
     sp_application_crypto::app_crypto!(sr25519, KEY_TYPE);
 }
 
+impl Default for SubstrateStorage {
+    fn default() -> Self {
+        let keystore = sc_keystore::LocalKeystore::in_memory();
+        Self {
+            inner: Arc::new(keystore),
+        }
+    }
+}
+
 impl SubstrateStorage {
+    /// Creates a new `SubstrateStorage` instance.
+    ///
+    /// This wrapper is used to provide a substrate-backed local key storage.
     #[must_use]
-    pub fn new(inner: sc_keystore::LocalKeystore) -> Self {
+    pub fn new(inner: Arc<sc_keystore::LocalKeystore>) -> Self {
```

Comment:
```suggestion
    pub fn new(inner: sc_keystore::LocalKeystore) -> Self {
```

Hide the `Arc` internally